src/snowflake/cli/_plugins/nativeapp/sf_sql_facade.py

```diff
@@ -206,3 +248,121 @@ def get_account_event_table(self, role: str | None = None) -> str | None:
+        @param [Optional] role: Role to switch to while running this script. Current role will be used if no role is passed in.
+        @param [Optional] database: Database to use while running this script.
+        @param [Optional] schema: Schema to use while running this script.
+        """
+        with (
+            self._use_role_optional(role),
+            self._use_database_optional(database),
+            self._use_schema_optional(schema),
+        ):
+            try:
+                results = self._sql_executor.show_specific_object(
+                    "stages",
+                    name,
+                    name_col=NAME_COL,
+                    check_schema=True,
+                )
+            except ProgrammingError as e:
+                if e.errno == DOES_NOT_EXIST_OR_CANNOT_BE_PERFORMED:
+                    return False
+                raise
+        return bool(results)
+
+    def create_stage(
+        self,
+        name: str,
+        encryption_type: str = "SNOWFLAKE_SSE",
+        enable_directory: bool = True,
+        role: str | None = None,
+        database: str | None = None,
+        schema: str | None = None,
+    ):
+        """
+        Creates a stage.
+        @param name: Name of the stage to check for. Can be a fully qualified name or just the stage name, in which case the current database and schema or the database and schema passed in will be used.
+        @param [Optional] encryption_type: Encryption type for the stage. Default is Snowflake SSE. Pass an empty string to disable encryption.
+        @param [Optional] enable_directory: Directory settings for the stage. Default is enabled.
+        @param [Optional] role: Role to switch to while running this script. Current role will be used if no role is passed in.
+        @param [Optional] database: Database to use while running this script.
+        @param [Optional] schema: Schema to use while running this script.
+        """
+        query = f"create stage if not exists {name}"
+        if encryption_type:
+            query += f" encryption = (type = '{encryption_type}')"
+        if enable_directory:
+            query += f" directory = (enable = {str(enable_directory)})"
+        with (
+            self._use_role_optional(role),
+            self._use_database_optional(database),
+            self._use_schema_optional(schema),
+        ):
+            try:
+                self._sql_executor.execute_query(query)
+            except ProgrammingError as e:
+                if e.errno == INSUFFICIENT_PRIVILEGES:
+                    raise InsufficientPrivilegesError(
+                        f"Insufficient privileges to create stage {name}",
+                        role=role,
+                        database=database,
+                        schema=schema,
+                    ) from e
+                raise
+
```

Comment:
make sure to use `handle_unclassified_error` for all the other exceptions that are caught from the facade.

Reply:
done